src/rust/client-executor/src/lib.rs

```diff
@@ -0,0 +1,287 @@
+pub mod action;
+use action::Action;
+
+use std::future::Future;
+use std::iter::{IntoIterator, Iterator};
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+use pin_project::pin_project;
+use tokio::time::{sleep_until, Duration, Instant, Sleep};
+use recloser::{AsyncRecloser, AnyError, RecloserFuture, Recloser, RecloserBuilder};
+
+pub use tokio_retry::strategy;
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error<E: std::error::Error> {
+    #[error("Reject")]
+    Rejected,
+    #[error(transparent)]
+    Inner(E)
+}
+
+#[pin_project(project = ExecuteStateProj)]
+enum ExecuteState<A>
+    where
+        A: Action,
+{
+    Running(#[pin] RecloserFuture<<A as Action>::Future, AnyError>),
+    Sleeping(#[pin] Sleep),
+}
+
+impl<A: Action> ExecuteState<A> {
+    fn poll(self: Pin<&mut Self>, cx: &mut Context) -> ExecuteFuturePoll<A> {
+        match self.project() {
+            ExecuteStateProj::Running(future) => {
+                match future.poll(cx) {
+                    Poll::Ready(Err(recloser::Error::Inner(e))) => ExecuteFuturePoll::Running(Poll::Ready(Err(Error::Inner(e)))),
+                    Poll::Ready(Err(recloser::Error::Rejected)) => ExecuteFuturePoll::Running(Poll::Ready(Err(Error::Rejected))),
```

Comment:
At first I was wondering why this crate's Error just mirrors recloser::Error instead of just using it with thiserror's `#[from] recloser::Error`, but since that'd be this crate's only variant it seems like you could just type alias recloser::Error instead? At least for now, until you add more variants you might have plans for. Thoughts?

Reply:
Ah, now I see recloser::Error doesn't impl std::error::Error and it doesn't require the E generic does either, as you do.

Reply:
`recloser::Error` does not implement `std::error::Error`, and I wanted to avoid re-exporting the crate.